docs/chapters/04_features.md

```diff
@@ -487,6 +487,60 @@ mutation {
 
 > [!NOTE]  Remember that, in case you want to subscribe to a read model that is restricted to a specific set of roles, you must send the **access token** retrieved upon sign-in. Check ["Authorizing operations"](#authorizing-operations) to know how to do this.
 
+
+### Adding "before" hooks to your read models
+
+When you send queries or subscriptions to your read models, you can tell Booster to execute some code before executing the operation. These are called `before` hooks, and they receive two parameters: the read model filter sent with the request and an object with information about the currently signed-in user.
+
+These `before` hooks are useful for many things, but mainly:
+- To add fine-grained access control. For example, to allow user `A` access to its Cart read models, but not the ones belonging to user `B`.
+
+- Modify the read model filters sent with the request based on some criteria. For example, you can narrow the results the users will get by adding extra filters based on the season of the year.
+  From inside a `before` hook, you can either throw an exception, in which case the query or subscription will be aborted, and the error will be sent back to the user, or return final filters to be used in the operation.
+  This is an example of how you would define a before hook in a read model:
+
+```typescript
+@ReadModel({
+  authorize: [User],
+  before: [CartReadModel.validateUser],
```

Comment:
Hmm the problem I see about this solution is user experience. To me is much easier to let the IDE autocomplete the function based on the class while typing. Maybe when I introduce the string, it's not right and I have to recheck if I mistyped it. Also, it simplifies how is passed to the core as it's a normal function

Reply:
One of the ideas of the before hooks is that you can put their functions that are defined anywhere. They can be functions in other modules of your project or functions that are in a third-party library. This is very powerful because we can publish a module with common before hooks or you can use a module with authorization hooks that is defined in a specific Auth rocket, for example.

The example that Juanjo used here happens to use static functions defined inside the read model class, but this is just a coincidence.

If we use strings, we would loose the ability to use any function.

Reply:
If its the case for using any kind of function then I'd rephrase the example with the hooks outside of the class instead of being static methods.

Reply:
I finally mixed the example with static and external methods 👯